src/main/java/org/grouplens/grapht/InjectionContainer.java

```diff
@@ -98,11 +100,11 @@ public Instantiator makeInstantiator(DAGNode<Component, Dependency> node,
         Instantiator cached = providerCache.get(node);
         if (cached == null) {
             logger.debug("Node has not been memoized, instantiating: {}", node.getLabel());
-            Map<Desire, Instantiator> depMap = makeDependencyMap(node, backEdges);
+            CachePolicy policy = node.getLabel().getCachePolicy();
 
-            Instantiator raw = node.getLabel().getSatisfaction().makeInstantiator(depMap);
+            Map<Desire, Instantiator> depMap = makeDependencyMap(node, backEdges);
 
-            CachePolicy policy = node.getLabel().getCachePolicy();
+            Instantiator raw = node.getLabel().getSatisfaction().makeInstantiator(depMap, this);
```

Comment:
It looks like the main change is adding the `this` parameter; is there a reason for re-ordering the other lines of code? It makes the diff harder to read.